Makefile

```diff
@@ -8,8 +8,9 @@ BUILD_CHANNEL ?= local
 PATH_WITH_TOOLS="`pwd`/$(TOOL_BIN):`pwd`/node_modules/.bin:${PATH}"
 
 GIT_REVISION = $(shell git rev-parse HEAD | tr -d '\n')
-TAG_VERSION?=$(shell git tag --points-at | sort -Vr | head -n1)
-LDFLAGS = -ldflags "-s -w -extld="$(shell pwd)/etc/ld_wrapper.sh" -X 'go.viam.com/rdk/config.Version=${TAG_VERSION}' -X 'go.viam.com/rdk/config.GitRevision=${GIT_REVISION}'"
+TAG_VERSION?=$(shell git tag --points-at | sort -Vr | head -n1 | grep . || echo "(dev)")
+DATE_COMPILED?=$(shell date +'%Y-%m-%d')
```

Comment:
[opt] A friend of mine mentioned that using the go debug `vcs.time` would work as well.

See:  https://pkg.go.dev/runtime/debug#BuildSetting
We report this in `viam --debug version`

Note: if you compile the cli with `go build ./viam/main.go` the vcs debug info won't be there. https://github.com/golang/go/issues/51279

I don't want to drag down this PR in refactors so no need to do this now, but might be worth considering in the future.